internal/core/translations.rs

```diff
@@ -305,25 +305,70 @@ pub fn translate_from_bundle_with_plural(
     output
 }
 
-pub fn set_language_internal(language: &str, languages: &[&str]) -> bool {
-    let idx = languages.iter().position(|x| *x == language);
-    if let Some(idx) = idx {
-        crate::context::GLOBAL_CONTEXT.with(|ctx| {
-            let Some(ctx) = ctx.get() else { return false };
+/// This function is called by the generated code to assign the list of bundled languages.
+/// Do nothing if the list is already assigned.
+pub fn set_bundled_languages(languages: &[&'static str]) {
+    crate::context::GLOBAL_CONTEXT.with(|ctx| {
+        let Some(ctx) = ctx.get() else { return };
+        if ctx.0.translations_bundle_languages.borrow().is_none() {
+            ctx.0.translations_bundle_languages.replace(Some(languages.to_vec()));
+        }
+    });
+}
+
+/// Select the current translation language when using bundled translations.
+/// This function only has effect if the slint code was compiled with bundled translations.
+/// It must be called after creating the first component.
+/// It returns an error if the language is not found.
+pub fn select_bundled_translation(locale: &str) -> Result<(), SelectBundledTranslationError> {
+    crate::context::GLOBAL_CONTEXT.with(|ctx| {
+        let Some(ctx) = ctx.get() else {
+            return Err(SelectBundledTranslationError::NoLanguageBundled);
+        };
+        let languages = ctx.0.translations_bundle_languages.borrow();
+        let Some(languages) = &*languages else {
+            return Err(SelectBundledTranslationError::NoLanguageBundled);
+        };
+        let idx = languages.iter().position(|x| *x == locale);
+        if let Some(idx) = idx {
             ctx.0.translations_dirty.as_ref().set(idx);
-            true
-        })
-    } else if language == "en" {
-        crate::context::GLOBAL_CONTEXT.with(|ctx| {
-            let Some(ctx) = ctx.get() else { return false };
+            Ok(())
+        } else if locale == "" || locale == "en" {
             ctx.0.translations_dirty.as_ref().set(0);
-            true
-        })
-    } else {
-        false
+            Ok(())
+        } else {
+            Err(SelectBundledTranslationError::LanguageNotFound {
+                available_languages: languages.iter().map(|x| (*x).into()).collect(),
+            })
+        }
+    })
+}
+
+/// Error type returned from the [`select_bundled_translation`] function.
+#[derive(Debug)]
+pub enum SelectBundledTranslationError {
+    /// The language was not found. Available languages are ...
+    LanguageNotFound { available_languages: crate::SharedVector<SharedString> },
+    /// There is no bundled languages. Either the select_bundled_translation was called to soon (before creating a component), or the generation was done without the bundle translation option)
```

Comment:
```suggestion
    /// There are no bundled languages. Either `select_bundled_translation` was called before creating a component, or the application's `.slint` file was compiled without the bundle translation option.
```

(There should really be a link to the build crate's docs here how to enable it...)